projects/ngx-d3timeline/src/lib/store/store.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { ActionTypes } from './actions';
+import { shareReplay, scan } from 'rxjs/operators';
+import { initialState } from './state';
+import { State } from './state';
+import { TimelineView } from '../view/timeline-view';
+import { Actions } from './actions';
+import { ScaleService } from '../scale.service';
+import { OptionsService } from '../options.service';
+
+@Injectable({ providedIn: 'root' })
+export class Store {
+  private readonly replayBuffer = 100;
```

Comment:
Could make it clearer what the buffer is for?

```suggestion
  private readonly bufferUntilSubscribed = 100;
```

Reply:
I think the buffer stays even when you unsubscribe. Maybe `replaySubjectBuffer`? 

The rxjs source just calls it buffer size
```
 constructor(bufferSize: number = Number.POSITIVE_INFINITY, windowTime: number = Number.POSITIVE_INFINITY, scheduler?: SchedulerLike)
```

Reply:
> I think the buffer stays even when you unsubscribe

Yep it will, maybe `bufferUntilSubscribed` is misleading.

The wider problem I was attempting to address, is that a ReplaySubject is used here but to the uninitiated there is no clue why.

`replaySubjectBuffer` is no more meaningful than `replayBuffer`.

`replayBufferSize` would be better than either of these even though adding 'size' makes it near impossible to convey the purpose of the buffer.

Any ideas how we can convey the meaning or should we just use a StoreSubject instead :grin:

Reply:
I'll name it `replayBufferSize`. 

> The wider problem I was attempting to address, is that a ReplaySubject is used here but to the uninitiated there is no clue why.

True but looking up the documentation would make it clear,